python/dolfinx/fem/petsc.py

```diff
@@ -1388,16 +1356,17 @@ def solve(self) -> tuple[PETSc.Vec, int, int]:  # type: ignore
         """
 
         # Move current iterate into the work array.
-        assign(self._u, self.x)
+        assign(self.u, self.x)
 
         # Solve problem
         self.solver.solve(None, self.x)
+        dolfinx.la.petsc._ghost_update(self.x, PETSc.InsertMode.INSERT, PETSc.ScatterMode.FORWARD)
```

Comment:
This ghost update was missing, but is present in `LinearProblem.solve`.

Reply:
Indeed there is a ghost update in the `assemble_residual` function. That one has always been there.